File: ruby/testcode/benchmark_test_00700.rb
require_relative 'shared'
require 'openssl'

SECRET = 'weak'

def sign_cookie(value)
  OpenSSL::HMAC.hexdigest('SHA256', SECRET, value)
end

def set_signed_cookie(req)
  token = req.param('token')
  sig = sign_cookie(token)
  response = BenchmarkResponse.ok('cookie set')
  response.headers['Set-Cookie'] = "session=#{token}.#{sig}; Secure; HttpOnly; SameSite=Strict"
  response
end

File: rust/testcode/benchmark_test_00700.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let a: i64 = req.param("a").parse().unwrap_or(0);
    let b: i64 = req.param("b").parse().unwrap_or(0);
    let sum = if 5 * 5 == 25 {
        a.saturating_add(b)
    } else {
        a + b
    };
    super::shared::BenchmarkResponse::ok(&format!("sum={}", sum))
}
